### PR TITLE
Constant error messages

### PR DESCRIPTION
**Pull request by mrLSD**
Set errors as constant messages with separated modules for crates

## Reason
* Avoid duplication of error messages and reduce binary size.
* Organize for more convenient way to manage error messages

### CODE REVIEW

File: engine/src/errors.rs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+pub const ERR_INVALID_ECDSA_SIGNATURE: &[u8; 27] = b"ERR_INVALID_ECDSA_SIGNATURE";
+pub const ERR_INTRINSIC_GAS: &[u8; 17] = b"ERR_INTRINSIC_GAS";
+pub const ERR_MAX_PRIORITY_FEE_GREATER: &[u8; 28] = b"ERR_MAX_PRIORITY_FEE_GREATER";
+pub const ERR_GAS_OVERFLOW: &[u8; 16] = b"ERR_GAS_OVERFLOW";
+pub const ERR_BALANCE_OVERFLOW: &[u8; 20] = b"ERR_BALANCE_OVERFLOW";
+pub const ERR_GAS_ETH_AMOUNT_OVERFLOW: &[u8; 27] = b"ERR_GAS_ETH_AMOUNT_OVERFLOW";
+pub const ERR_PARSE_ADDRESS: &[u8; 17] = b"ERR_PARSE_ADDRESS";
+pub const ERR_STATE_NOT_FOUND: &[u8; 19] = b"ERR_STATE_NOT_FOUND";
+pub const ERR_STATE_CORRUPTED: &[u8; 19] = b"ERR_STATE_CORRUPTED";
+
```
Comment (joshuajbouw) [inline engine/src/errors.rs:53]: Can you fix the spacing on all these please?

**Review by joshuajbouw — CHANGES_REQUESTED** [review]

**Comment by sept-en** [comment]
As the refactoring is being made, IMO it'd be good to separate these errors into submodules: e.g. `errors::evm::ERR_OUT_OF_GAS`, `errors:json::ERR_NOT_A_JSON_TYPE`, etc.

WDYT @joshuajbouw ?

**Comment by mrLSD** [comment]
> As the refactoring is being made, IMO it'd be good to separate these errors into submodules: e.g. `errors::evm::ERR_OUT_OF_GAS`, `errors:json::ERR_NOT_A_JSON_TYPE`, etc.
> 
> WDYT @joshuajbouw ?

@sept-en AFAIK main idea was deduplication.
Your proposal is good as a logical separation. But we can't guarantee in that case deduplication error messages.

After splitting error messages as separated module `errors` we almost do not have benefits with bin size.
But after that, a lot of error messages were deduplicated. And I think it is good.

**Comment by Aurora-ClickUp** [comment]
Task linked: [CU-2g5nvuc The binary size can be reduced by using constants for error messages.](https://app.clickup.com/t/2g5nvuc)

**Comment by joshuajbouw** [comment]
@sept-en I think its a good idea for the future, but not right now. We got to focus on some more higher importance matters.

**Review by joshuajbouw — APPROVED** [review]

**Comment by joshuajbouw** [comment]
Please fix conflicts.

**Comment by sept-en** [comment]
Created #547 to ensure we track possible future refactoring somewhere

**Pull request merged**